### PR TITLE
chore: Add JSLibrary autocomplete analytics

### LINKED ISSUE #25281: [Task]: Track usage of custom JS libraries
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

Description: Track the usage of objects and methods that belong to a custom JS library 

Event: `AUTO_COMPLETE_SELECT`

Additional information to be tracked on the payload: 
```
{
    "isExternalLibrary": <boolean>
    "libraryNamespace": <string>
}
```

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
#### PR fixes following issue(s)
Fixes #25281

#### Type of change

- Chore (housekeeping or task changes that don't impact user perception)

- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/utils/autocomplete/CodemirrorTernService.ts [hunk 1]
```diff
@@ -399,6 +399,8 @@ class CodeMirrorTernService {
         selectedResultType: selected.type,
         isBestMatch:
           selectedResultIndex <= AutocompleteSorter.bestMatchEndIndex,
+        isExternalLibrary: selected.origin?.startsWith("LIB/"),
```
Comment (arunvjn) [inline app/client/src/utils/autocomplete/CodemirrorTernService.ts:402]: This is will track all library including the defaults like moment, lodash etc. Is that the intent ?

Comment (Rishabh-Rathod) [reply app/client/src/utils/autocomplete/CodemirrorTernService.ts:402]: @bharath31 Do we want to have a differentiate between default JSLibrary usage and installed library usage in autocomplete?

Comment (bharath31) [reply app/client/src/utils/autocomplete/CodemirrorTernService.ts:402]: @Rishabh-Rathod @arunvjn tracking all library usage(default + custom) should be helpful

**Pull request merged**